workflow/rules/align.smk

```diff
@@ -43,7 +43,15 @@ rule Bowtie2_Pathogenome_Alignment:
     benchmark:
         "benchmarks/BOWTIE2/{sample}.benchmark.txt"
     message:
-        "Bowtie2_Pathogenome_Alignment: ALIGNING SAMPLE {input.fastq} TO PATHOGENOME WITH BOWTIE2"
+        "Bowtie2_Alignment: ALIGNING SAMPLE {input.fastq} WITH BOWTIE2"
     shell:
-        """bowtie2 --large-index -x {params.PATHO_DB} --end-to-end --threads {threads} --very-sensitive -U {input.fastq} 2> {log} | samtools view -bS -q 1 -h -@ {threads} - | samtools sort - -@ {threads} -o {output.bam} >> {log};"""
-        """samtools index {output.bam}"""
+        """bowtie2 --large-index -x {params.BOWTIE2_DB} --end-to-end --threads {threads} --very-sensitive -U {input.fastq} > $(dirname {output.bam})/AlignedToBowtie2DB.sam 2> {log};"""
+        """grep @ $(dirname {output.bam})/AlignedToBowtie2DB.sam | awk '!seen[$2]++' > $(dirname {output.bam})/header_nodups.txt;"""
```

Comment:
Since samtools is used further down, why not replace `grep @` by `samtools head` to view the header, and `grep -v '^@'` below to `samtools view --no-header`? This would make it easier to understand the logic of the commands. 

Are all the commands here meant to rewrite the samtools header to remove duplicates? Why not use `samtools reheader`? Speed issues? This shouldn't be the case; see http://www.htslib.org/doc/samtools-reheader.html, where the description reads "This command is much faster than replacing the header with a BAM→SAM→BAM conversion. "

Reply:
See issue #122